In [50]:
import pandas as pd
import numpy as np
import csv
import time
import yaml

In [21]:
start = time.time()

df_pd = pd.read_csv('Books_rating.csv')

end = time.time()
print(end - start)

28.311389207839966


In [23]:
start = time.time()

data = []
with open("Books_rating.csv", "r") as csvfile:
    reader_variable = csv.reader(csvfile, delimiter=",")
    for row in reader_variable:
        data.append(row)

end = time.time()
print(end - start)

48.513229846954346


In [41]:
df_pd['Title'] = df_pd['Title'].apply(lambda x: ''.join(filter(str.isalpha, str(x))).lower())
df_pd.head()

Id                    Title  Price         User_id  \
0  1882931173  itsonlyartifitswellhung    NaN   AVCGYZL8FQQTD   
1  0826414346      drseussamericanicon    NaN  A30TK6U7DNS82R   
2  0826414346      drseussamericanicon    NaN  A3UH4UZ4RSVO82   
3  0826414346      drseussamericanicon    NaN  A2MVUWT453QH61   
4  0826414346      drseussamericanicon    NaN  A22X4XUPKF66MR   

                          profileName review/helpfulness  review/score  \
0               Jim of Oz "jim-of-oz"                7/7           4.0   
1                       Kevin Killian              10/10           5.0   
2                        John Granger              10/11           5.0   
3  Roy E. Perry "amateur philosopher"                7/7           4.0   
4     D. H. Richards "ninthwavestore"                3/3           4.0   

   review/time                                   review/summary  \
0    940636800           Nice collection of Julie Strain images   
1   1095724800                                Really Enjoyed It   
2   1078790400  Essential for every personal and Public Library   
3   1090713600  Phlip Nel gives silly Seuss a serious treatment   
4   1107993600                           Good academic overview   

                                         review/text  
0  This is only for Julie Strain fans. It's a col...  
1  I don't care much for Dr. Seuss but after read...  
2  If people become the books they read and if "t...  
3  Theodore Seuss Geisel (1904-1991), aka &quot;D...  
4  Philip Nel - Dr. Seuss: American IconThis is b...

In [43]:
df_pd.columns

Index(['Id', 'Title', 'Price', 'User_id', 'profileName', 'review/helpfulness',
       'review/score', 'review/time', 'review/summary', 'review/text'],
      dtype='object')

In [44]:
%%writefile file.yaml
file_type: csv
dataset_name: Amazon Review 
file_name: Books_rating
inbound_delimiter: ","
outbound_delimiter: "|"
skip_leading_rows: 1
columns:
  - Id
  - Title
  - Price
  - User_id
  - profileName
  - review/helpfulness
  - review/score
  - review/time
  - review/summary
  - review/text


Writing file.yaml


In [54]:
with open('file.yaml') as f:
    my_dict = yaml.safe_load(f)
my_dict['columns']

['Id',
 'Title',
 'Price',
 'User_id',
 'profileName',
 'review/helpfulness',
 'review/score',
 'review/time',
 'review/summary',
 'review/text']

In [57]:
import gzip

text_data = "\n".join("|".join(row) for row in data)

with gzip.open('output_file.gz', 'wt') as f:
    f.write(text_data)

In [58]:
import os

In [60]:
size_bytes = os.path.getsize('output_file.gz')
size_bytes

1054180627

In [62]:
df_pd.shape

(3000000, 10)

# Summary

### Number of Columns: 10
### Number of Rows: 3000000
### gz File Size: 1054180627 bytes